lib/libwebsockets.c

```diff
@@ -510,14 +510,28 @@ lws_close_free_wsi(struct lws *wsi, enum lws_close_status reason)
 	    !wsi->socket_is_permanently_unusable) {
 #ifdef LWS_OPENSSL_SUPPORT
 		if (lws_is_ssl(wsi) && wsi->ssl)
-               {
-                       lwsl_info("%s: shutting down SSL connection: %p (ssl %p, sock %d, state %d)\n", __func__, wsi, wsi->ssl, (int)(long)wsi->desc.sockfd, wsi->state);
+		{
+			lwsl_info("%s: shutting down SSL connection: %p (ssl %p, sock %d, state %d)\n", __func__, wsi, wsi->ssl, (int)(long)wsi->desc.sockfd, wsi->state);
 			n = SSL_shutdown(wsi->ssl);
-                       if (n == 1) /* If finished the SSL shutdown, then do socket shutdown, else need to retry SSL shutdown */
-                               n = shutdown(wsi->desc.sockfd, SHUT_WR);
-                       else
-                               lws_change_pollfd(wsi, LWS_POLLOUT, LWS_POLLIN);
-               }
+			if (n == 1) /* If finished the SSL shutdown, then do socket shutdown, else need to retry SSL shutdown */
+				n = shutdown(wsi->desc.sockfd, SHUT_WR);
+			else if (n == 0)
+				lws_change_pollfd(wsi, LWS_POLLOUT, LWS_POLLIN);
```

Comment:
Guys going by the description in this issue, the root cause is we don't detect the peer has gone (by SYSCALL / errno).  All other sequencing is meaningless  that situation if that's the case.

If it's not that the peer has dropped, it won't busyloop, because it's the false POLLIN indicating the peer dropped that drives that.

If it's not interesting to confirm SYSCALL with a specific errno let me know how to reproduce the problem in the issue with the test server and I'll look at it.